File: api/src/main/java/domain/collaborators/enterprise/entities/Company.java
package domain.collaborators.enterprise.entities;

import core.entities.Entity;

import java.util.Date;
import java.util.UUID;

public class Company extends Entity {

    private final UUID id;
    private final String cnpj;
    private final String name;
    private final Date createdAt;

    private Company(UUID id, String cnpj, String name, Date createdAt) {
        this.id = id;
        this.cnpj = cnpj;
        this.name = name;
        this.createdAt = createdAt;
    }

    public UUID getId() {
        return this.id;
    }

    public String getCnpj() {
        return this.cnpj;
    }

    public String getName() {
        return this.name;
    }

    public Date getCreatedAt() {
        return this.createdAt;
    }

    public static Company create(String cnpj, String name) {
        UUID id = UUID.randomUUID();
        Date createdAt = new Date();

        return new Company(id, cnpj, name, createdAt);
    }

    public static Company create(UUID id, String cnpj, String name, Date createdAt) {

        return new Company(id, cnpj, name, createdAt);
    }

}

File: api/src/domain/users/enterprise/entities/company.py
from datetime import datetime
from uuid import UUID, uuid4

from src.core.entities.entity import Entity
from src.domain.users.dto.create_company_request_dto import CreateCompanyRequestDTO


class Company(Entity):
    def __init__(self, id: UUID, cnpj: str, name: str, created_at: datetime):
        self.id = id
        self.cnpj = cnpj
        self.name = name
        self.created_at = created_at

    @staticmethod
    def create(props: CreateCompanyRequestDTO) -> 'Company':
        if props.id is None:
            props.id = uuid4()

        if props.created_at is None:
            props.created_at = datetime.now()

        company = Company(props.id, props.cnpj, props.name, props.created_at)

        return company

    def to_dict(self) -> dict:
        return {
            "id": str(self.id),
            "cnpj": self.cnpj,
            "name": self.name,
            "created_at": self.created_at
        }